code/modules/mob/living/carbon/xenomorph/castes/baneling/baneling.dm

```diff
@@ -0,0 +1,88 @@
+/mob/living/carbon/xenomorph/baneling
+	caste_base_type = /mob/living/carbon/xenomorph/baneling
+	name = "Baneling"
+	desc = ""
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Walking"
+	bubble_icon = "alienleft"
+	health = 100
+	maxHealth = 100
+	plasma_stored = 50
+	tier = XENO_TIER_ONE
+	upgrade = XENO_UPGRADE_ZERO
+	pixel_x = -16
+	old_x = -16
+
+/// We do this to avoid a runtime with images assoc because baneling goes inside of their pod if they have one and there is no atom.z value inside of the pod
+/mob/living/carbon/xenomorph/baneling/on_death()
+	if(!isnull(pod_ref))
+		return
+	return ..()
+
+/// Delete the pod if we evolve or devolve
+/mob/living/carbon/xenomorph/baneling/finish_evolve()
+	if(!isnull(pod_ref))
+		pod_ref.baneling_ref = null
+		pod_ref.obj_destruction()
+		pod_ref = null
+	return ..()
+
+/obj/structure/xeno/baneling_pod
+	name = "Baneling Pod"
+	desc = "A baneling pod, storing fresh banelings "
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Pod"
+	density = FALSE
+	obj_flags = CAN_BE_HIT | PROJ_IGNORE_DENSITY
+	/// Maximum amount of stored charge
+	var/stored_charge_max = 2
+	/// Respawn charges, each charge makes respawn take 30 seconds. Maximum of 2 charges. If there is no charge the respawn takes 120 seconds.
+	var/stored_charge = 0
+	/// How long until we get another charge
+	var/charge_refresh_time = 180 SECONDS
+	/// Time to respawn if we have charges
+	var/respawn_time = 30 SECONDS
+	/// Ref to our baneling
+	var/mob/living/carbon/xenomorph/baneling/baneling_ref
```

Comment:
Should be usable for any xeno that gets given the pod action, same with the rest of the file.